### PR TITLE
pkp/pkp-lib#935: replace multiple LEFT OUTER JOINs in Review Forms

### PR DESCRIPTION
**Pull request by ctgraham**
Change LEFT OUTERs to EXISTS / NOT EXISTS clauses if needed; remove one unused LEFT OUTER JOIN set.  This is done to improve performance on large sets.

Resolves https://github.com/pkp/pkp-lib/issues/935

### CODE REVIEW

**Comment by beghelli** [comment]
Hi @ctgraham ,

I am no expert, but a quick research and I saw this: http://explainextended.com/2009/09/18/not-in-vs-not-exists-vs-left-join-is-null-mysql/ and http://explainextended.com/2009/09/16/not-in-vs-not-exists-vs-left-join-is-null-postgresql/

If I got it correct and according to that material, when you use LEFT JOIN together with IS NULL or  IS NOT NULL on the query, both MySQL and Postgres are able to optimize that, and the performance is the same or even faster than EXISTS. Were you able to test and see the performance improvement? Thanks.

**Comment by ctgraham** [comment]
I'm testing against MySQL 5.1, as the default version shipped on RHEL.  The difference in performance with NOT EXISTS was similar to the [dramatic difference on NOT IN](https://github.com/pkp/pkp-lib/issues/935#issue-119522899).

File: classes/reviewForm/ReviewFormDAO.inc.php [hunk 1]
```diff
@@ -278,17 +278,7 @@ function &getActiveByAssocId($assocType, $assocId, $rangeInfo = null) {
...
-					rac.review_form_id = rf.review_form_id AND
-					rac.date_confirmed IS NOT NULL
-				)
-				LEFT JOIN review_assignments rai ON (
-					rai.review_form_id = rf.review_form_id AND
-					rai.date_notified IS NOT NULL AND
-					rai.date_confirmed IS NULL
-				)
 			WHERE	rf.assoc_type = ? AND assoc_id = ? AND rf.is_active = 1
-			GROUP BY rf.assoc_type, rf.assoc_id, rf.review_form_id, rf.seq, rf.is_active
```
Comment (beghelli) [inline classes/reviewForm/ReviewFormDAO.inc.php]: maybe you forgot to add the EXISTS sql here?

Comment (ctgraham) [inline classes/reviewForm/ReviewFormDAO.inc.php]: No, this removal is intentional.  See https://github.com/pkp/pkp-lib/issues/935#issuecomment-160763433

All these joins do is slow the query!  :-)

Comment (beghelli) [inline classes/reviewForm/ReviewFormDAO.inc.php]: :) great. @asmecher, do you want to test with postgres this?

Comment (asmecher) [inline classes/reviewForm/ReviewFormDAO.inc.php]: It's a trivial enough query -- I ran a quick smoke test:

    SELECT rf.* FROM review_forms rf WHERE rf.assoc_type = 1 AND assoc_id = 1 AND rf.is_active = 1 ORDER BY rf.seq;

...and it looks OK to me.

**Pull request merged**